05_syntax_lab/04.py

```diff
@@ -1,8 +1,14 @@
-"""
-Write a program that reads lines from the user
-until an empty line is inserted.
-After the user typed in an empty line,
-print all previously inserted lines in reverse
-order (from last to first)
-"""
-
+################## Part 4
+text = []
+while 1:
+    print('Please enter a string')
+    tmp = input()
+    if len(tmp) <= 0:
+        break
+    text.append(tmp)
+    #print(text)
+
+i = len(text)
+while i > 0:
+    print(text[i-1])
+    i -= 1

```

Comment:
* It is better to use ```True``` instead of 1 in the infinite while loop. That's the real ```True``` of Python.
* ```input()``` can have a parameter which is the prompt text so you don't need to print separately.